ASIM/dev/ASimTester/Validate-ASimCsv/Validate-ASimCsv.ps1

```diff
@@ -0,0 +1,50 @@
+
+[CmdletBinding()]
+param (
+    [Parameter(Mandatory, ValueFromPipeline)]
+    [ValidateScript(
+        {
+            (Test-Path -Path $_) -and ($_.Extension -in '.csv')
+        }
+    )]
+    [System.IO.FileInfo]$FilesPath
+)
+
+$requiredColumns = @(
+    'ColumnName',
+    'ColumnType',
+    'Class',
+    'Schema',
+    'LogicalType',
+    'ListOfValues',
+    'Aliased'
+)
+
+$csv = Get-ChildItem -Path $FilesPath | Import-Csv -Delimiter ','
+$columns = (($csv | Select-Object -First 1).PSObject.Members | Where-Object MemberType -eq 'NoteProperty').name
+
+if (-not(Compare-Object -ReferenceObject $requiredColumns -DifferenceObject $columns)) {
+    Write-Debug "The file '$($FilesPath.FullName)' is valid"
+}
+else {
+    Write-Warning "The file '$($FilesPath.BaseName)' is not valid"
+    break
+}
+
+$schemas = $csv | Select-Object -ExpandProperty Schema -Unique
+$duplicateColumns = @()
+
+foreach ($schema in $schemas) {
+    Write-Debug "Processing ASIM schema '$($schema)'"
+    $schemaArray = ($csv | Where-Object schema -eq $schema)
+
+    foreach ($item in $schemaArray) {
+        $count = ($schemaArray | Where-Object ColumnName -eq $item.ColumnName).count
+        if ($count -gt 1) {
+            Write-Debug "Found duplicate column '$($item.ColumnName)' in schema '$($schema)'"
+            $duplicateColumns += $item
+        }
+    }
+}
+
+return $duplicateColumns | Sort-Object Schema, ColumnName, Class -Unique
```

Comment:
When duplicate fields exists, can we add a message at the top that describes the issue? Duplicate entries found for the following fields.
Also, when no duplicate entries, it should say no duplicate entries exists. csv is good. and exit

Reply:
### When the file contains no duplicates
![image](https://github.com/Azure/Azure-Sentinel/assets/40334679/3468e656-5da3-4789-8e79-733bdd4fb237)

### When the file contains duplicates
![image](https://github.com/Azure/Azure-Sentinel/assets/40334679/b08db452-0641-44c5-a09e-997c142bf9db)